mockwebserver/src/main/java/okhttp3/mockwebserver/MockResponse.java

```diff
@@ -254,6 +257,21 @@ public long getBodyDelay(TimeUnit unit) {
   }
 
   /**
+   * Set the time to {@code delay} prior to shutting down an HTTP/2 connection. This is only valid
+   * with {@link SocketPolicy#GRACEFUL_SHUTDOWN_AT_END}. Use this to simulate a graceful HTTP/2
+   * connection shutdown.
+   */
+  public MockResponse setShutdownDelay(long delay, TimeUnit unit) {
+    shutdownDelay = delay;
+    shutdownDelayUnit = unit;
+    return this;
+  }
+
+  public long getShutdownDelay(TimeUnit unit) {
```

Comment:
... the upside of just storing milliseconds in the field is that accessors don’t need to do this awkward thing, where precision can be lost.

... though this _is_ consistent with java.util.concurrent...
https://docs.oracle.com/javase/7/docs/api/java/util/concurrent/Delayed.html#getDelay(java.util.concurrent.TimeUnit)